web/packages/teleterm/src/ui/ModalsHost/modals/AuthenticateWebDevice/AuthenticateWebDevice.tsx

```diff
@@ -24,22 +24,22 @@ import { useAsync } from 'shared/hooks/useAsync';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import { RootClusterUri, routing } from 'teleterm/ui/uri';
-
-type Props = {
-  rootClusterUri: RootClusterUri;
-  onCancel(): void;
-  onClose(): void;
-  onAuthorize(): Promise<void>;
-};
+import { retryWithRelogin } from 'teleterm/ui/utils';
 
 export const AuthenticateWebDevice = ({
   onAuthorize,
   onClose,
   onCancel,
   rootClusterUri,
-}: Props) => {
+}: {
+  rootClusterUri: RootClusterUri;
+  onCancel(): void;
+  onClose(): void;
+  onAuthorize(): Promise<void>;
+}) => {
+  const ctx = useAppContext();
   const [attempt, run] = useAsync(async () => {
-    await onAuthorize();
+    await retryWithRelogin(ctx, rootClusterUri, onAuthorize);
```

Comment:
I don't think this works. After my cert expires, clicking the button launches the Web UI anyway (that's with `disconnect_expired_cert: yes`).

https://github.com/user-attachments/assets/6f0d32e1-1d87-45d7-b7bb-e9d330197e25